cx-portal/src/components/pages/AppSubscription/ActivateSubscriptionOverlay/index.tsx

```diff
@@ -0,0 +1,312 @@
+ * Copyright (c) 2021, 2023 BMW Group AG
+ * Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogHeader,
+  Input,
+  LoadingButton,
+  Typography,
+} from 'cx-portal-shared-components'
+import HelpOutlineIcon from '@mui/icons-material/HelpOutline'
+import { useTranslation, Trans } from 'react-i18next'
+import Patterns from 'types/Patterns'
+import CheckCircleOutlinedIcon from '@mui/icons-material/CheckCircleOutlined'
+import {
+  SubscriptionActivationResponse,
+  useAddUserSubscribtionMutation,
+  useFetchTechnicalProfilesQuery,
+} from 'features/appSubscription/appSubscriptionApiSlice'
+import { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import './style.scss'
+import { store } from 'features/store'
+import { setSuccessType } from 'features/appSubscription/slice'
+import { Link } from 'react-router-dom'
+
+const TentantHelpURL =
+  '/documentation/?path=docs%2F04.App%28s%29%2F05.+App-Subscription%2F04.+Subscription+Activation%28App+Provider%29.md'
+const ProfileHelpURL =
+  '/documentation/?path=docs%2F04.App%28s%29%2F05.+App-Subscription%2F04.+Subscription+Activation%28App+Provider%29.md'
+
+interface ActivateSubscriptionProps {
+  openDialog: boolean
+  appId: string
+  subscriptionId: string
+  title: string
+  handleOverlayClose: () => void
+}
+
+const ActivateSubscriptionOverlay = ({
+  openDialog = false,
+  appId,
+  subscriptionId,
+  title,
+  handleOverlayClose,
+}: ActivateSubscriptionProps) => {
+  const { t } = useTranslation()
+  const dispatch = useDispatch<typeof store.dispatch>()
+  const [inputURL, setInputURL] = useState('')
+  const [URLErrorMsg, setURLErrorMessage] = useState('')
+  const [loading, setLoading] = useState(false)
+  const [activationResponse, setActivationResponse] =
+    useState<SubscriptionActivationResponse>()
+
+  const [addUserSubscribtion] = useAddUserSubscribtionMutation()
+  const { data } = useFetchTechnicalProfilesQuery(appId)
+
+  const addInputURL = (value: string) => {
+    setInputURL(value)
+    setURLErrorMessage(
+      !Patterns.URL.test(value.trim())
```

Comment:
we have a more expressive function for this:
`isURL = (expr: string) => Patterns.URL.test(expr)`
https://github.com/eclipse-tractusx/portal-frontend/blob/dev/cx-portal/src/types/Patterns.ts#L93